src/psyclone/domain/nemo/transformations/create_nemo_invoke_schedule_trans.py

```diff
@@ -0,0 +1,159 @@
+                                Reference[name:'ji']
+                            BinaryOperation[operator:'MUL']
+                                Literal[value:'2.0', Scalar<REAL, UNDEFINED>]
+                                Reference[name:'ji']
+
+    The root node of this example has been transformed from a Routine into a
+    NemoInvokeSchedule.
+
+    """
+    @property
+    def name(self):
+        '''
+        :returns: the name of the transformation.
+        :rtype: str
+
+        '''
+        return type(self).__name__
+
+    def validate(self, node, options=None):
+        '''
+        Check that the supplied node is a valid target for this transformation.
+
+        :param node: the target of the transformation.
+        :type node: :py:class:`psyclone.psyir.nodes.Node`
+        :param options: a dictionary with options for \
+            transformations. No options are used in this \
+            transformation. This is an optional argument that defaults \
+            to None.
+        :type options: dict of string:values or None
+
+        :raises TransformationError: if the supplied node is not a Routine.
+
+        '''
+        super(CreateNemoInvokeScheduleTrans, self).validate(node,
+                                                            options=options)
+
+        if not isinstance(node, Routine):
+            raise TransformationError(
+                "Error in NemoInvokeTrans transformation. The supplied node "
+                "should be a PSyIR Routine but found '{0}'".format(
+                    type(node).__name__))
+
+    def apply(self, routine, options=None):
+        '''
+        Takes a generic PSyIR Routine and replaces it with a
+        NemoInvokeSchedule (in-place). Note that this may mean replacing
+        the top-level node itself and therefore this routine returns the
+        root of the modified tree.
+
+        :param routine: the routine node to be transformed.
+        :type routine: :py:class:`psyclone.psyir.nodes.Routine`
+        :param options: a dictionary with options for \
+            transformations. No options are used in this \
+            transformation. This is an optional argument that defaults \
+            to None.
+        :type options: dict of string:values or None
+
+        :returns: the new PSyIR node that replaces the Routine.
+        :rtype: :py:class:`psyclone.nemo.NemoInvokeSchedule`
+
+        '''
+        self.validate(routine, options=options)
+
+        new_node = NemoInvokeSchedule.create(routine.name,
+                                             routine.symbol_table,
+                                             routine.pop_all_children(),
+                                             is_program=routine.is_program)
+
+        # We need to replace the top node in the (possibly sub-) PSyIR
+        # tree that we've been passsed.
+        if routine.parent:
+            routine.replace_with(new_node)
+        del routine
+
+        return new_node
```

Comment:
This is different that the Transformation generic method that returns `root, memento`. I know that we are not consistent on the apply interface (which we should fix at some point), but in this case couldn't we always return the root node as in the generic interface `return new_node.root, None`?

Reply:
We have issue #595 for this. There I argued that transformations shouldn't return anything but given that this particular transformation might change the root node of the whole tree I'm not sure we can do that anymore. Having said that, I've seen mention of a `FileContainer` node so maybe we can argue that a `Routine` will never be the root of the whole tree? If we do that though then a lot of the NEMO tests will need updating as they often work from a string that just contains a program or subroutine definition.

Anyway, for now I've followed you suggestion and it now returns the suggested tuple.